compiler+runtime/src/cpp/jank/runtime/core/math.cpp

```diff
@@ -516,20 +529,11 @@ namespace jank::runtime
   {
     return visit_number_like(
       [](auto const typed_l, auto const r) -> object_ptr {
-        using L = typename decltype(typed_l)::value_type;
-
         return visit_number_like(
           [](auto const typed_r, auto const typed_l) -> object_ptr {
-            using R = typename decltype(typed_r)::value_type;
-
-            if constexpr(std::is_same_v<L, obj::real> || std::is_same_v<R, obj::real>)
-            {
-              return make_box(std::fmod(typed_l, typed_r->data));
-            }
-            else
-            {
-              return make_box(typed_l % typed_r->data);
-            }
+            auto typed_l_data = to_number(typed_l);
+            auto typed_r_data = to_number(typed_r->data);
```

Comment:
Please use direct initialization everywhere and make everything const.

```cpp
            auto const typed_l_data{ to_number(typed_l) };
            auto const typed_r_data{ to_number(typed_r->data) };
```

Reply:
updated.